lib/virtio/virtqueue.c

```diff
@@ -586,6 +681,12 @@ static int vq_ring_enable_interrupt(struct virtqueue *vq, uint16_t ndesc)
 #endif /*VIRTIO_MASTER_ONLY*/
 	}
 
+#ifdef VIRTIO_D_CACHE
+	metal_cache_flush(VRING_USED(vq), vring_used_size(VRING_NUM(vq)));
+	metal_cache_flush(VRING_AVAIL(vq),
```

Comment:
only master should fetch the VRING_AVAIL and only slave should flush the VRING_USED, else you ca create a memory corruption.
same issue for the virtqueue_disable_cb function

Reply:
ACK